### PR TITLE
MetadataStore - handle only editing and saving, remove duplicity

### LINKED ISSUE #2439: Duplicate metadata store
https://github.com/keboola/kbc-ui/blob/14621a616d24b8cc7b6c55816ead2a40ccef9087/src/scripts/modules/components/stores/MetadataStore.js#L133-L145

^ I think this means, that whole `metadata` part from `MetadataStore` is duplication of what we alraedy have in `StorageTablesStore`

### PR DESCRIPTION
**Pull request by natocTo**
Fixes #2439 - or at least most of it

I removed everything except storing editing value and information about saving metadata.

Metadata is taken direct from a bucket or a table. Updated metadata is stored back to a bucket or a table.

MetadataStore was used:
- when editing description (bucket, table) - still used
- in transformation input mapping (snowflake) - metadata is read direct from table

### CODE REVIEW

File: src/scripts/modules/components/react/components/MetadataEditField.jsx [hunk 1]
```diff
@@ -1,66 +1,72 @@
...
+import MetadataStore from '../../stores/MetadataStore';
 
 export default React.createClass({
-
-  displayName: 'MetadataEditField',
-
   mixins: [createStoreMixin(MetadataStore), immutableMixin],
 
   propTypes: {
-    objectType: PropTypes.oneOf(['bucket', 'table', 'column']).isRequired,
```
[the remark this answers is not in the document]
Comment (pivnicek) [reply src/scripts/modules/components/react/components/MetadataEditField.jsx]: You're right, there aren't any yet, but there should be! :-)  Column descriptions for example

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx [hunk 2]
```diff
@@ -256,6 +250,41 @@ export default React.createClass({
     });
   },
 
+  getTableLastUpdatedInfo(table) {
```
[the remark this answers is not in the document]
Comment (natocTo) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: I looked where this function was used and the only place was it this component. So I thought it is perfect spot. :) But I can definitely move it somewhere.

File: src/scripts/modules/components/react/components/MetadataEditField.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (pivnicek) [reply src/scripts/modules/components/react/components/MetadataEditField.jsx]: I'm actually looking forward to chatting with you about UI use of metadata at offsite :-)

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (pivnicek) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: I think the component and config that create/update the table would be useful information for table details no?

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx [hunk 3]
```diff
@@ -256,6 +250,41 @@ export default React.createClass({
...
+    }
+
+    return {
+      'component': componentFound.get('value'),
+      'config': configFound.get('value'),
+      'timestamp': configFound.get('timestamp')
+    };
+  },
+
+  hasTableMetadataDatatypes(selectedTable) {
```
[the remark this answers is not in the document]
Comment (pivnicek) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: for example, writers will want to use this too.

[the remark this answers is not in the document]
Comment (pivnicek) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: and storage info I think.

File: src/scripts/modules/components/react/components/MetadataEditField.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (natocTo) [reply src/scripts/modules/components/react/components/MetadataEditField.jsx]: My communication skills is far from perfect even in Czech, I hope I will be able to talk in english: 👍 (no experience but I try my best)

**Comment by natocTo** [comment]
Updated.

File: src/scripts/modules/components/react/components/MetadataEditField.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (natocTo) [reply src/scripts/modules/components/react/components/MetadataEditField.jsx]: "column" is back :)

[the remark this answers is not in the document]
Comment (pivnicek) [reply src/scripts/modules/components/react/components/MetadataEditField.jsx]: neboje, muzem spolu muvlit czechish nebo englishtina :-D

**Comment by pivnicek** [comment]
@natocTo @ujovlado what is the status of this PR?  I'm working on a little poc metadata editor but it probably makes sense to wait for this one?

**Comment by natocTo** [comment]
Waiting for review.

**Comment by pivnicek** [comment]
The changes since my last review are 👍 .  only there is a conflict now.  @ujovlado I think will want to look at it, but if not I can do a fresh check/monkey check.

**Comment by natocTo** [comment]
Tady je to samý konflikt no, tak to zkusím trošku použít jako inspiraci a pak to smažu. Nebo pokud z toho nic nevykoukám tak to smažu navečer.

**Pull request closed without merging**